bpf/lib/host_firewall.h

```diff
@@ -185,8 +185,55 @@ ipv6_host_policy_ingress(struct __ctx_buff *ctx, __u32 *src_id)
 # endif /* ENABLE_IPV6 */
 
 # ifdef ENABLE_IPV4
+#  ifndef ENABLE_MASQUERADE
 static __always_inline int
-ipv4_host_policy_egress(struct __ctx_buff *ctx, __u32 src_id)
+whitelist_snated_egress_connections(struct __ctx_buff *ctx, __u32 ipcache_srcid)
+{
+	struct ct_state ct_state_new = {}, ct_state = {};
+	struct ipv4_ct_tuple tuple = {};
+	void *data, *data_end;
+	struct iphdr *ip4;
+	__u32 monitor = 0;
+	int ret, l4_off;
+
+	/* If kube-proxy is in use (no BPF-based masquerading), packets from
```

Comment:
But this also gets enabled for kube-proxt-free w/o BPF masq, no? What is the relationship there?

Reply:
Hm. If we are running kube-proxy-free without BPF masquerading, the ipcache lookup won't resolve the security ID of packets from pods as `HOST_ID`, will it?

We briefly discussed this with Martynas and I couldn't find a better check for this than the `#ifndef ENABLE_MASQUERADE` above :-/